repository_service_tuf_api/__init__.py

```diff
@@ -97,60 +100,70 @@ def sync_redis():
+            with open(secrets_settings.TOKEN_KEY) as f:
+                SECRET_KEY = f.read().rstrip("\n")
+        except OSError as err:
+            logging.error(str(err))
 
-# User database setup
-DATABASE_URL = settings.get(
-    "DATABASE_URL", f"sqlite:///{os.path.join(DATA_DIR, 'users.sqlite')}"
-)
-
-engine = create_engine(DATABASE_URL, connect_args={"check_same_thread": False})
-SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+    else:
+        SECRET_KEY = secrets_settings.TOKEN_KEY
 
+    if secrets_settings.ADMIN_PASSWORD.startswith("/run/secrets/"):
+        try:
+            with open(secrets_settings.ADMIN_PASSWORD) as f:
+                ADMIN_PASSWORD = f.read().rstrip("\n")
+        except OSError as err:
+            logging.error(str(err))
 
-Base.metadata.create_all(bind=engine)
-db = SessionLocal()
+    else:
+        ADMIN_PASSWORD = secrets_settings.ADMIN_PASSWORD
 
-for scope in SCOPES:
-    scope_entry = crud.get_scope_by_name(db, name=scope)
-    if not scope_entry:
-        add_scope = schemas.ScopeCreate(name=scope, description=SCOPES[scope])
-        crud.create_user_scope(db, add_scope)
+    # User database setup
+    DATABASE_URL = settings.get(
+        "DATABASE_URL", f"sqlite:///{os.path.join(DATA_DIR, 'users.sqlite')}"
+    )
 
-user = crud.get_user_by_username(db, username="admin")
-if not user:
-    user_in = schemas.UserCreate(
-        username="admin",
-        password=ADMIN_PASSWORD,
+    engine = create_engine(
+        DATABASE_URL, connect_args={"check_same_thread": False}
     )
-    user = crud.create_user(db, user_in)
-    crud.user_add_scopes(db, user, [scope for scope in crud.get_scopes(db)])
+    SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+    Base.metadata.create_all(bind=engine)
+    db = SessionLocal()
 
-else:
-    user_scopes = [user_scope.name for user_scope in user.scopes]
     for scope in SCOPES:
-        if scope not in user_scopes:
-            crud.user_append_scope(db, user, scope)
-            logging.info(f"Scope '{scope}' added to admin.")
+        scope_entry = crud.get_scope_by_name(db, name=scope)
+        if not scope_entry:
+            add_scope = schemas.ScopeCreate(
+                name=scope, description=SCOPES[scope]
+            )
+            crud.create_user_scope(db, add_scope)
+
+    user = crud.get_user_by_username(db, username="admin")
+    if not user:
+        user_in = schemas.UserCreate(
+            username="admin",
+            password=ADMIN_PASSWORD,
+        )
+        user = crud.create_user(db, user_in)
+        crud.user_add_scopes(
+            db, user, [scope for scope in crud.get_scopes(db)]
+        )
+
+    else:
+        user_scopes = [user_scope.name for user_scope in user.scopes]
+        for scope in SCOPES:
+            if scope not in user_scopes:
+                crud.user_append_scope(db, user, scope)
+                logging.info(f"Scope '{scope}' added to admin.")
```

Comment:
We haven't added a test for that before, right?
Do we want to?

Reply:
For now, we won't fix this.
See: https://github.com/repository-service-tuf/repository-service-tuf-api/pull/329#discussion_r1182549739